var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -222,12 +232,11 @@ class Dealii(CMakePackage, CudaPackage):
     # do not require +rol to make concretization of xsdk possible
     depends_on("trilinos+amesos+aztec+epetra+ifpack+ml+muelu+sacado", when="+trilinos")
     depends_on("trilinos~hypre", when="+trilinos+int64")
-    # TODO: temporary disable Tpetra when using CUDA due to
-    # namespace "Kokkos::Impl" has no member "cuda_abort"
-    depends_on(
-        "trilinos@master+rol~amesos2~ifpack2~intrepid2~kokkos~tpetra~zoltan2",
-        when="+trilinos+cuda",
-    )
+    for a in CudaPackage.cuda_arch_values:
+        arch_str = "+cuda cuda_arch=" + a
+        trilinos_spec = "trilinos +wrapper " + arch_str
```

Comment:
(minor) Can you use an f-string to make this a bit more readable? Also, it's maybe better to use a different name instead of `a` to avoid name clash with other variables. Maybe start it with `_`.